### DISCUSSION TITLE
请问如果外文资料的书面翻译希望翻译一篇arXiv的话，文章号出不来怎么办？

### DISCUSSION THREAD

**Discussion by yuchenguommm** [tuna/thuthesis#828] [Q&A] [answered]
如下
![image](https://user-images.githubusercontent.com/88889094/216738494-ca1d4107-6bdb-4559-be48-c2ccaa39782b.png)

正文的参考文献中的arXiv们是可以正常显示文章号的。
其实这个问题是说，如何在附录中也使用和正文同样的参考文献格式
将\bibliographystyle{unsrtnat}更换成\bibliographystyle{thuthesis-bachelor}并没有效果

**Comment by yuchenguommm** [comment]
已通过错误的方法暂时解决：从正文的bbl里面复制了这一条并且手动做了一个bbl文件

求问正确的做法中，这一部分的文献格式应该如何调整？

**Reply by zepinglee** [reply]
@yc-guo19 把这条文献的 .bib 内容贴上来看一下？

> 已通过错误的方法暂时解决：从正文的bbl里面复制了这一条并且手动做了一个bbl文件
> 
> 求问正确的做法中，这一部分的文献格式应该如何调整？

手动做的 bbl 格式也贴上来。

**Reply by yuchenguommm** [reply]
这条文献的.bib如下：
```
@Misc{Torlai2020,
  author        = {Giacomo Torlai and Christopher J. Wood and Atithi Acharya and Giuseppe Carleo and Juan Carrasquilla and Leandro Aolita},
  title         = {Quantum process tomography with unsupervised learning and tensor networks},
  year          = {2020},
  archiveprefix = {arXiv},
  eprint        = {2006.02424},
  primaryclass  = {quant-ph},
}
```
手动制作的bbl如下
```
\begin{thebibliography}{1}
\providecommand{\natexlab}[1]{#1}
\providecommand{\url}[1]{#1}
\expandafter\ifx\csname urlstyle\endcsname\relax\else
  \urlstyle{same}\fi
\expandafter\ifx\csname href\endcsname\relax
  \DeclareUrlCommand\doi{\urlstyle{rm}}
  \def\eprint#1#2{#2}
\else
  \def\doi#1{\href{https://doi.org/#1}{\nolinkurl{#1}}}
  \let\eprint\href
\fi

\bibitem[Torlai et~al.(2020)Torlai, Wood, Acharya, Carleo, Carrasquilla, and
  Aolita]{Torlai2020}
TORLAI G, WOOD C~J, ACHARYA A, et~al.
\newblock Quantum process tomography with unsupervised learning and tensor
  networks\allowbreak[A].
\newblock 2020.
\newblock arXiv: \eprint{https://arxiv.org/abs/2006.02424}{2006.02424}.

\end{thebibliography}
```
其中前一半是直接从正文编译生成的bbl文件中拷贝过来的。

**Reply by yuchenguommm** [reply]
事实上，这一问题在附录的参考文献部分普遍存在，我认为这是由于模板中附录的引用格式是unsrt，而不是正文的thuthesis-bachelor；但我不知如何调整。将\bibliographystyle{unsrtnat}更换成\bibliographystyle{thuthesis-bachelor}并没有效果。

**Reply by zepinglee** [reply]
> 将\bibliographystyle{unsrtnat}更换成\bibliographystyle{thuthesis-bachelor}并没有效果。

我在本地的编译结果如下。

```tex
% !TeX root = ../thuthesis-example.tex

\begin{translation}
\label{cha:translation}

Foo\cite{Torlai2020}.

% 书面翻译的参考文献
\bibliographystyle{thuthesis-bachelor}
\bibliography{ref/appendix}

\end{translation}
```

<img width="660" alt="Screenshot 2023-02-07 at 12 07 46" src="https://user-images.githubusercontent.com/12290822/217145932-702a0335-0972-49f5-bbb8-c524af69eb2e.png">

@yc-guo19 你修改的是这一行吗？

https://github.com/tuna/thuthesis/blob/9f934e108005bb2b119683abc017d5d153340965/data/appendix-translation.tex#L73

**Reply by yuchenguommm** [reply]
是的 @zepinglee

**Reply by zepinglee** [reply, marked as answer]
上面的截图中，替换为 `\bibliographystyle{thuthesis-bachelor}` 可以输出 arxiv 文章编号，跟正文的参考文献样式相同。
